### DISCUSSION TITLE
Where are the Unit Tests in 8.2.0?

### DISCUSSION THREAD

**Discussion by davehouser1** [Kong/insomnia#6677] [Help] [answered]
I tried installing 8.2.0, even logged in with a cloud account. In 2023.5.8 there was a "test" button at the top of Insomnia. I don't see this in 8.2.0. I don't see it in the scratch pad either.
Your website [here ](https://insomnia.rest/product/automated-testing) shows its still a feature. 
Is this not the case? Are unit tests gone, or am I missing something?

**Comment by subnetmarco** [comment, marked as answer]
You need to create a Design document, then click on the "Test" tab. This is for creating tests for your APIs.

<img width="1728" alt="Screenshot 2023-10-19 at 10 22 49 PM" src="https://github.com/Kong/insomnia/assets/964813/fb770ac6-03e6-4a63-9ad7-d13908fee224">

Or are you looking at something more similar to the concept of a "Collection" run in Postman?

**Reply by davehouser1** [reply]
Ah there it is very good. I also see that you all implemented displaying the verb for each request for the test, a needed enhancement. 
Thank you for the details. 
Do you have any idea when an update will be made to support the following?

- Support for tests in scratch pad. 
- Support duplicating tests suite.
- Support duplicating unit tests in a suite.
- Support re-arraigning unit tests in a suite.
- Support for async requests in a given test.
- Support for access to environmental variables in tests.
- Support to show total time all tests in a suite took to run.

**Reply by subnetmarco** [reply]
We are going to be starting a new sprint focused on tests sometime by EoY, so this is great feedback. We were also thinking to decouple tests from the design spec view, so that it's easier to find them and manage them, like this:

<img width="475" alt="Screenshot 2023-10-20 at 10 23 18 AM" src="https://github.com/Kong/insomnia/assets/964813/926fd6ba-03d9-41b2-a3aa-3a24c30cfd79">

You can also see that we are planning to also decouple environments and authentications so they can be re-used across files, including in test files.

Scratch Pad will not innovate much in the future besides improving the HTTP client (collection debug) capability. All the other capabilities, including the ones in the screenshot above, will be available in the Insomnia dashboard.

**Reply by davehouser1** [reply]
That's good news! I would say access to environmental variables in unit tests and rearrange / duplicate unit tests would be the most needed features (imo). I made a discussion post / comment on the async issues I was having if that helps you all.
I have to say, your replies and news of these upcoming features for tests is exciting and makes me want to continue to use Insomnia!
